libminifi/include/utils/GeneralUtils.h

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file GeneralUtils.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef LIBMINIFI_INCLUDE_UTILS_GENERAL_UTILS_H
+#define LIBMINIFI_INCLUDE_UTILS_GENERAL_UTILS_H
+
+#include <memory>
+#include <type_traits>
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+template<typename T, typename... Args>
+std::unique_ptr<T> make_unique(Args&&... args) {
+  return std::unique_ptr<T>{ new T{ std::forward<Args>(args)... } };
+}
+
+template<typename T, typename std::enable_if<std::is_integral<T>::value>::type* = nullptr>
+T intdiv_ceil(T numerator, T denominator) {
+  // note: division and remainder is 1 instruction on x86
+  return numerator / denominator + (numerator % denominator > 0);
+}
+
+template <typename T, typename std::enable_if<std::is_pointer<T>::value>::type* = nullptr>
```

Comment:
This construct is good for marking that the owner of a raw pointer has the responsibility of deallocating it, but I would argue, that aside from interfacing with C code, this is the only case a raw pointer should be used in C++ anyway.
I think in the case of `rd_kafka_headers_s` creating a RAII wrapper for it would be a better solution.

Reply:
`std::unique_ptr<rd_kafka_headers_t, decltype([](rd_kafka_headers_t* p) { rd_kafka_headers_destroy(p); })>` both acts as an RAII wrapper and does the null checking.

p.s.: can't wait for C++20